exporter/datadogexporter/go.mod

```diff
@@ -10,7 +10,7 @@ require (
 	github.com/DataDog/datadog-agent/comp/logs/agent/config v0.56.2
 	github.com/DataDog/datadog-agent/comp/otelcol/logsagentpipeline v0.56.2
 	github.com/DataDog/datadog-agent/comp/otelcol/logsagentpipeline/logsagentpipelineimpl v0.56.2
-	github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/logsagentexporter v0.56.2
+	github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/logsagentexporter v0.58.0-devel.0.20240905201012-b02662b2ba27
```

Comment:
cc @mackjmr please use the same version in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35051